### ISSUE TITLE
Suricata container

### ISSUE THREAD

**Issue by mattmac1** [dtag-dev-sec/t-pot-autoinstall#11]
Hi,
I created an issue under tpot but closed that and creating here.

I have used the autoinstall script to install tpot standard on amazon EC2. Everything works except there is no suricata container - the image is there though. If i check Portainer I can see 8 containers but 9 images.

Installed on 64 bit ubuntu 16.04 and with enough resources so 4 Gb memory, 64 Gb storage.

As you would expect docker-status shows:

======| System |======
Date:     Mon Feb 20 11:34:01 UTC 2017
Uptime:   11:34:01 up 46 min, 1 user, load average: 0.01, 0.10, 0.06
CPU temp:

======| Container: cowrie |======
cowrie                           RUNNING    pid 8, uptime 0:46:36
ewsposter                        RUNNING    pid 10, uptime 0:46:36
mysqld                           RUNNING    pid 9, uptime 0:46:36

======| Container: dionaea |======
dionaea                          RUNNING    pid 10, uptime 0:46:36
ewsposter                        RUNNING    pid 9, uptime 0:46:36

======| Container: elasticpot |======
elasticpotpy                     RUNNING    pid 9, uptime 0:46:36

======| Container: elk |======
elasticsearch                    RUNNING    pid 14, uptime 0:46:36
kibana                           RUNNING    pid 16, uptime 0:46:36
logstash                         RUNNING    pid 15, uptime 0:46:36

======| Container: glastopf |======
ewsposter                        RUNNING   pid 10, uptime 0:46:36
glastopf                         RUNNING   pid 17, uptime 0:46:34

======| Container: honeytrap |======
ewsposter                        RUNNING    pid 8, uptime 0:46:38
honeytrap                        RUNNING    pid 9, uptime 0:46:38

======| Container: suricata |======

**Comment by ip999** [comment]
Hi,

I found the same issue (same scenario, running on ubuntu in AWS). I'm not too familiar with suricata but I found that the error was being caused by the ethtool command to disable tx and rx-checksumming. I commented out this and the next line for segmentation offload too for good measure (but not tested to see if that was really necessary).

comment out these two lines in suricata.service:

#ExecStartPre=/bin/bash -c '/sbin/ethtool --offload $(/sbin/ip route | /bin/grep $(/bin/hostname -I | /usr/bin/awk \'{print $1 }\') | /usr/bin/awk \'{print $3 }\') rx off tx off'
#ExecStartPre=/bin/bash -c '/sbin/ethtool -K $(/sbin/ip route | /bin/grep $(/bin/hostname -I | /usr/bin/awk \'{print $1 }\') | /usr/bin/awk \'{print $3 }\') gso off gro off'

It seems to work for me and suricata is now started and seeing traffic - not sure of any implications but it suited my basic needs for the time being.

**Comment by t3chn0m4g3** [comment]
@ip999 This is strange since the command should read (notice the `\`'s):
```
ExecStartPre=/bin/bash -c '/sbin/ethtool --offload $(/sbin/ip route | /bin/grep $(/bin/hostname -I | /usr/bin/awk \'{print $1 }\') | /usr/bin/awk \'{print $3 }\') rx off tx off'
ExecStartPre=/bin/bash -c '/sbin/ethtool -K $(/sbin/ip route | /bin/grep $(/bin/hostname -I | /usr/bin/awk \'{print $1 }\') | /usr/bin/awk \'{print $3 }\') gso off gro off'
```
Can you please retry? TIA!

**Comment by ip999** [comment]
Here was the systemctl status before it was commented out:

[ubuntu@dresserapple:/etc/init.d]$ systemctl status suricata.service
● suricata.service - suricata
   Loaded: loaded (/etc/systemd/system/suricata.service; enabled; vendor preset: enabled)
   Active: inactive (dead) (Result: exit-code) since Tue 2017-02-28 21:01:30 UTC; 33s ago
  Process: 16006 ExecStartPre=/bin/bash -c /sbin/ethtool --offload $(/sbin/ip route | /bin/grep $(/bin/hostname -I | /usr/bin/awk '{print $1 }') | /usr/bin/awk '{print $3 }') rx off tx off (code=exited, status=1/FAILURE)

**Comment by t3chn0m4g3** [comment]
Can you please retry with the ones I posted above? Want to know if that fixes it. If offloading is not deactivated the checksums might be garbage and hence Suricata might detect more false / positives.

**Comment by ip999** [comment]
ah- looks like they got lost in a copy+paste somewhere... the `\`'s are there. 

To confirm i'm not going mad... tried again below with formatting..

.# = original lines from my suricata.service  
.$ = the lines you pasted above...

```
#ExecStartPre=/bin/bash -c '/sbin/ethtool --offload $(/sbin/ip route | /bin/grep $(/bin/hostname -I | /usr/bin/awk \'{print $1 }\') | /usr/bin/awk \'{print $3 }\') rx off tx off'
$ExecStartPre=/bin/bash -c '/sbin/ethtool --offload $(/sbin/ip route | /bin/grep $(/bin/hostname -I | /usr/bin/awk \'{print $1 }\') | /usr/bin/awk \'{print $3 }\') rx off tx off'
#ExecStartPre=/bin/bash -c '/sbin/ethtool -K $(/sbin/ip route | /bin/grep $(/bin/hostname -I | /usr/bin/awk \'{print $1 }\') | /usr/bin/awk \'{print $3 }\') gso off gro off'
$ExecStartPre=/bin/bash -c '/sbin/ethtool -K $(/sbin/ip route | /bin/grep $(/bin/hostname -I | /usr/bin/awk \'{print $1 }\') | /usr/bin/awk \'{print $3 }\') gso off gro off'
```

**Comment by t3chn0m4g3** [comment]
LGTM, works fine on 16.04.2 in VM.

What is the result if you run the commands on its own?

**Comment by t3chn0m4g3** [comment]
Found the root cause, will be fixed in 17.06.
You can find the details here dtag-dev-sec/tpotce/issues/99
